### PR TITLE
Issue #400

### PR DESCRIPTION
**Pull request by victoriadanyelcortes**
This should now modify listings in place to prevent the sidebar from reloading.

### CODE REVIEW

File: web/src/app/sidebar/interested-courses/component.ts [hunk 1]
```diff
@@ -55,8 +58,25 @@ export class InterestedCoursesComponent implements OnInit {
         .includes('sections.listing')
         .includes('course')
         .includes('course.subject')
-        .all().then((listings) => {
-          this.listings = listings.data;
+        .all().then((newListings) => {
+          let oldListingsSet =  new Set(this.listings.map((listing) => listing.id));
+          let newListingsSet = new Set(newListings.data.map((newListing) => newListing.id));
+          for (let i in newListings.data) {
```
Comment (Bad-Science) [inline web/src/app/sidebar/interested-courses/component.ts]: Prefer `forEach` for `for..of`

File: web/src/app/sidebar/interested-courses/component.ts [hunk 2]
```diff
@@ -55,8 +58,25 @@ export class InterestedCoursesComponent implements OnInit {
...
+            let inCurList: boolean  = oldListingsSet.has(newListings.data[i].id);
+            if (!inCurList) {
+              this.listings.push(newListings.data[i]);
+
```
Comment (Bad-Science) [inline web/src/app/sidebar/interested-courses/component.ts]: remove extra newlines

File: web/src/app/sidebar/interested-courses/component.ts [hunk 3]
```diff
@@ -55,8 +58,25 @@ export class InterestedCoursesComponent implements OnInit {
...
+            }
+          }
+
+          for( let j = 0; j < this.listings.length; ++j) {
```
Comment (Bad-Science) [inline web/src/app/sidebar/interested-courses/component.ts]: same as above

**Comment by Bad-Science** [comment]
@victoriadanyelcortes 
Thanks for the PR! Left some small suggestions

**Comment by victoriadanyelcortes** [comment]
Finished the changes, hopefully eveything looks good!

**Pull request merged**